### ISSUE TITLE
translate.py gives me KeyError : None on this line corpus_opts = opts.tasks[corpus_id]

### ISSUE THREAD

**Issue by shakibyzn** [Helsinki-NLP/mammoth#45]
### System Info
env: nvcr.io_nvidia_pytorch_23.08-py3.sqsh Enroot image
python=3.10
pytorch=2.1

### Information
 ```
# job.sh file
#!/bin/bash

#SBATCH --job-name=test
#SBATCH --partition=A100-40GB
#SBATCH --time=2-0:00:00
#SBATCH --output out/slurm-%j.out
#SBATCH --nodes=1
#SBATCH --ntasks=1
#SBATCH --gres=gpu:4
#SBATCH --mem=64G

export MAMMOTH=`pwd`/mammoth
export CONFIG_DIR=`pwd`
export model=`pwd`/models/europarl/models/example-1-node_step_1000_frame.pt
# translate test set
srun \
    --container-image=/netscratch/enroot/nvcr.io_nvidia_pytorch_23.08-py3.sqsh \
    --container-workdir="`pwd`" \
    --container-mounts=/netscratch:/netscratch,/ds:/ds:ro,"`pwd`":"`pwd`" \
    --task-prolog="`pwd`/install.sh" \
    wrapper.sh -u ${MAMMOTH}/translate.py \
        -config ${CONFIG_DIR}/europarl-1node-4gpu.yml \
        -model ${model} \
        -src `pwd`/test.src \
        -tgt `pwd`/test.tgt \

```

```
# wrapper.sh file
#! /bin/bash

export CUDA_VISIBLE_DEVICES=0,1,2,3
nvidia-smi dmon -s mu -d 5 -o TD > "${LOG_DIR}/gpu_load-${EXP_ID}-${PPID}.log" &
echo python -u "$@" --node_rank $SLURM_NODEID
python -u "$@" --node_rank $SLURM_NODEID
```

### Tasks
 An officially supported translation example as step 5: translate in Training Mammoth 101

### Reproduction
Hi, I followed training Mammoth 101 and trained smoothly for 1000 steps. I wanted to translate a file with only one sentence:
test.src (Ich bin John)
test.tgt (I am John)
I ran into KeyError: None on this line ``` corpus_opts = opts.tasks[corpus_id] ``` in mammoth/mammoth/bin/translate.py file

### Expected behavior
Anyone can help?

**Comment by Waino** [comment]
Hi Shakib!

Thank you for your interest in Mammoth.

Try adding `--task_id train_de-en` to the arguments of `translate.py`. It is necessary to specify which of the tasks should be applied in translation. This defines the source and target languages, and which modules are loaded.

The error message is quite cryptic, and we should improve it by validating the parameter in the translation script.

**Comment by shakibyzn** [comment]
Hi Waino!

Thank you for your reply. I figured out that I need to set the `--task_id train_de-en` as you said. However, I'm dealing with another error in model_builder.py
```
mammoth/mammoth/model_builder.py", line 189, in load_test_model
model = checkpoint['whole_model']
KeyError: 'whole_model'
```
It seems that I need to change this part `export model=`pwd`/models/europarl/models/example-1-node_step_1000_frame.pt `. 

Could you help me with this?

**Comment by Waino** [comment]
Try `export model=`pwd`/models/europarl/models/example-1-node_step_1000`.

Mammoth normally outputs the checkpoint in a format where each module is in a separate `.pt` file.
The different suffixes of each module are appended to the given model path.
However, if you give a path directly to a `.pt` file, then it tries to load the whole model from a single file.

**Comment by shakibyzn** [comment]
Thanks a lot. It worked🙂

**Comment by Waino** [comment]
Original issue fixed in #47
